node/node.go

```diff
@@ -257,27 +260,40 @@ func createAndStartEventBus(logger log.Logger) (*types.EventBus, error) {
 	return eventBus, nil
 }
 
-func createAndStartIndexerService(config *cfg.Config, dbProvider DBProvider,
-	eventBus *types.EventBus, logger log.Logger) (*txindex.IndexerService, txindex.TxIndexer, error) {
+func createAndStartIndexerService(
+	config *cfg.Config,
+	dbProvider DBProvider,
+	eventBus *types.EventBus,
+	logger log.Logger,
+) (*txindex.IndexerService, txindex.TxIndexer, indexer.BlockIndexer, error) {
+
+	var (
+		txIndexer    txindex.TxIndexer
+		blockIndexer indexer.BlockIndexer
+	)
 
-	var txIndexer txindex.TxIndexer
 	switch config.TxIndex.Indexer {
 	case "kv":
```

Comment:
note to reviewer: we piggy-back off of the existing `tx` indexing configuration.

Reply:
Should this be piggybacked? It seems that if the node operator decides to index txs they won't have a way to avoid indexing blocks? 

Should we add a boolean to the config under the Indexing section for indexing blocks or not?

Reply:
See my initial question https://github.com/tendermint/tendermint/pull/6226#issuecomment-797067570.

Personally, I think we _should_. I can't think of a valid case or situation where you'd want to index block events but not tx events or vice versa.

Reply:
I agree with this choice.

Basically if you index one, you need to index both. Otherwise, you only get a partial image of the changes. The motivation was many effects occured in begin and endblock and were unsearchable. If we decouple them, that is likely to continue to be the case. I see this as extending tx indexer to be complete.

In my ideal world, they would be behind the same endpoint like subscriptions that capture both tx and end/begin block events. But I do see the purpose of separating them for the different info returned.

Reply:
Agree 👍